src/sentry/deletions/base.py

```diff
@@ -173,6 +181,10 @@ def chunk(self, num_shards=None, shard_id=None):
         """
         query_limit = self.query_limit
         remaining = self.chunk_size
+        has_more = True
+        # TODO(mark) This loop results in all of the work being done in
+        # one chunk/task. This loop would need to be removed if we want
+        # smaller chunks of work.
```

Comment:
I don't quite understand this comment. It looks like `remaining` is just `chunk_size`, so if we have more than `chunk_size` rows we'll exit the loop

Reply:
You're right I overlooked the `or` coalescing in the `__init__`, and was further distracted by tasks that only delete a single row returning true when there was no additional work to be done.